### PR TITLE
Disable reset practice board button when in starting FEN

### PR DESCRIPTION
**Pull request by rovmug-ticfyn**

### CODE REVIEW

File: src/frontend/tsx/Games.tsx
Comment (ashelkovnykov) [inline src/frontend/tsx/Games.tsx]: 1. The default FEN already exists as a constant in `ts/constants/chess`
2. This is an incredibly inefficient way to determine if the button should be enabled or not. This is doing a string comparison every single time the state updates, which means it needs to compare every single character of the current FEN string to every single character of the default FEN string. The better way to do this would be to add a variable like `hasGame` named `practiceHasMoved` or something like that. When the practice board is reset, `practiceHasMoved` is set to `false`. If the user makes a move, `practiceHasMoved` is set to `true`. While `practiceHasMoved` is `false`, the "reset practice board" button is disabled.

**Comment by bonbud-macryg** [comment]
@rovmug-ticfyn @ashelkovnykov The way this is implemented right now will conflict with #58, which moves the Practice Board buttons to the Game Panel.

**Comment by rovmug-ticfyn** [comment]
@bonbud-macryg I’ll wait to refactor until the game panel is merged

**Comment by bonbud-macryg** [comment]
@rovmug-ticfyn The skeleton of the game panel has been merged so you can refactor this whenever you're ready to.

We've talked in Design Notes about the buttons being icons rather than text like "reset practice board", but almost all of your refactored code should still apply there. Something to be aware of, but we shouldn't need to worry about those new buttons till the final UI tweaks.

**Review by ashelkovnykov — CHANGES_REQUESTED** [review]
- `practiceHasMoved` and `setPracticeHaveMoved` shouldn't be in `ChessState`. They should probably be [in the React state, with a copy in `localStorage`](https://www.freecodecamp.org/news/how-to-use-localstorage-with-react-hooks-to-set-and-get-items/).
  - For that matter, the FEN state of the practice board should also be in `localStorage`. This would allow us to store the practice board state between looking at games and different browsing sessions.
- The "Reset Practice Board" button should be enabled/disabled based on `practiceHasMoved`, and hidden/visible based on `hasGame`

You should put these changes at the bottom of the priority stack, since they're not directly linked to any Milestone 1 issue.

**Review by ashelkovnykov — APPROVED** [review]
LGTM. I didn't check locally, so I won't hit the merge button. Anyone who's verified it's working locally (including yourself @rovmug-ticfyn) can go ahead and merge.

**Comment by bonbud-macryg** [comment]
Happy to test it on my machine shortly, will merge if it's all good.

**Comment by rovmug-ticfyn** [comment]
Not working as expected. Hold off on merging for now.

**Comment by bonbud-macryg** [comment]
@rovmug-ticfyn Will wait until your say-so before I test it.

**Comment by rovmug-ticfyn** [comment]
My latest commit causes the button to work as expected (disabled when the practice board is in starting position, enabled otherwise), but it does so by comparing the FEN string against the default each time the practice board is saved. I'm setting this aside for now and will come back to work on a more efficient solution.

File: src/frontend/tsx/GamePanel.tsx
Comment (ashelkovnykov) [inline src/frontend/tsx/GamePanel.tsx]: Change this to:
```
const practiceHasMoved = (localStorage.getItem('practiceBoard') !== CHESS.defaultFEN)
```

File: src/frontend/tsx/GamePanel.tsx
Comment (ashelkovnykov) [inline src/frontend/tsx/GamePanel.tsx]: Should be both the old condition and the new one:
```
disabled={hasGame || !practiceHasMoved}
```

File: src/frontend/tsx/Chessboard.tsx
Comment (ashelkovnykov) [inline src/frontend/tsx/Chessboard.tsx]: You can drop all of the changes to this file

**Review by ashelkovnykov — CHANGES_REQUESTED** [review]
Alright, so Chessground not being a React component makes this a burden. We basically have two options:
1. Do this the "right"/efficient way and have a separate boolean that records whether or not pieces on the practice board has moved. Another annoying thing is that changes to `localStorage` do not force a React re-render, so you need to do something like [what's outlined in this guide](https://www.tutorialspoint.com/localstorage-in-reactjs).
2. Just do the thing that I didn't want to do originally and that you're doing now: check if the FEN for the board is the initial FEN. This works every time, but it's inefficient/"wrong". However, dealing with it is so much easier that we should probably do it and write this off as one of those stupid things that is the cost of using JS (`~hastuc-dibtyx`'s Urbit-native UI can't come soon enough...).

**Review by ashelkovnykov — APPROVED** [review]
LGTM

**Pull request merged**
